docs/source/extension/extension_points.rst

```diff
@@ -457,16 +457,180 @@ The recommended ranges for this rank are:
 Main Menu
 ---------
 
-There are three main ways to extend JupyterLab's main menu.
+There are two ways to extend  JupyterLab's main menu.
+
+1. Using the settings - this is the preferred way as they are configurable by the user.
+2. Using the API - this is for advanced cases like dynamic menu or semantic items.
+
+Settings-defined menu
+^^^^^^^^^^^^^^^^^^^^^
+
+JupyterLab provides integration with its settings system for menu definitions.
+Your extension can provide a settings schema with a ``jupyter.lab.menus`` key,
+declaring default menus. You don't need to set anything in the typescript code
+(except the command definitions).
```

Comment:
> except the command definitions

aw, man... we're getting so close! Pretty off-topic, but with the prior art of:

- OpenAPI `operationId`
- a way to reference DI tokens that are `provides`ed by plugins

...i feel like one can _almost_ confect at least simple things from just data... or at least advertise them

Reply:
I'm not following you on this one. But happy to have more of those data backed approach.